tests/integration_tests/clouds.py

```diff
@@ -300,8 +290,7 @@ def _perform_launch(self, launch_kwargs, **kwargs):
         except KeyError:
             profile_list = self._get_or_set_profile_list(release)
 
-        prefix = datetime.datetime.utcnow().strftime("cloudinit-%m%d-%H%M%S")
-        default_name = prefix + "".join(
+        default_name = self.cloud_instance.tag + "".join(
             random.choices(string.ascii_lowercase + string.digits, k=8)
```

Comment:
Now we have to "moving parts" in the instance naming: the timestamp set in pycloudlib and this random string. In Jenkins we may want to use fixed instance names to prevent the proliferation of stale instances should we have bugs in the cleanup code. This would require disabling both this random string and the pycloudlib timestamp.

I was thinking of dropping this random string bit, leaving it to pycloudlib to create unique instance names based on the tag. Then we can add a `pycloudlib.toml` setting to make it not add the timestamp in the cases where we want fixed instance names instead.

We already have seconds resolution in the pycloudlib timestamping, hopefully that's enough to prevent collisions. Thoughts?

Reply:
Yes this makes sense. If we have parallel jobs running in jenkins, it's likely they'll be different Ubuntu series or platforms so collisions should not generally occur. If we end up runing jobs for a single platform-series in parallel instead of sequentially at some point we can grow random chars at that point if we actually hit a collision problem. I'll drop this from this branch and we can incorporate that into pycloudlib in the future if needed.

Reply:
+1 on incorporating this into pycloudlib if needed.

Reply:
Had to delete my prior comment on this. I agree with the sentiement about pycloudlib randomizing instance name based on the cloud.tag prefix. The issue we have at the moment in cloud-init CI is that pycloudlib doesn't yet have this feature, and the tag is created only at the start of integration tests. So, until pycloudlib grows this feature to create names based on tags, this branch should either go to WIP/draft and await our work on pycloudlib, or add this random string to allow sequentially running mulitple integration tests in sequence. I'm good with WIP/draft on this PR if we want to add the pycloudlib feature first.